ow_materials/include/AxisAlignedGrid.h

```diff
@@ -0,0 +1,63 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef AXIS_ALIGNED_GRID_H
+#define AXIS_ALIGNED_GRID_H
+
+#include <vector>
+#include <memory>
+
+#include <ignition/math/AxisAlignedBox.hh>
+
+namespace ow_materials {
+
+template <typename T>
+class AxisAlignedGrid {
```

Comment:
Are there existing data structures that could be used here, such as an OpenCV Mat? Some investigation might reveal useful features you did not think to add to this class.

Reply:
In addition to OpenCV Mat there is Boost MultiArray. We would want to do interpolation (weighted averages of 8 voxels), but it doesn't appear that either of these data structures would help with that. It might still be worth evaluating these libraries for performance and API convenience.